iphone/Classes/TiLogServer.m

```diff
@@ -0,0 +1,347 @@
+/**
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2009-2016 by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+
+/**
+ * Important: Do NOT call NSLog() from this file. The app will go into an
+ * infinite loop of death. Use forcedNSLog() instead.
+ */
+
+#ifndef DISABLE_TI_LOG_SERVER
+
+#import "TiLogServer.h"
+#import "TiBase.h"
+#include <sys/socket.h>
+#include <netinet/in.h>
+
+/**
+ * These constants are available to allow xcodebuild to override them.
+ */
+#ifndef TI_LOG_SERVER_PORT
+# define TI_LOG_SERVER_PORT 10571
+#endif
+
+#ifndef TI_LOG_SERVER_BACKLOG
+# define TI_LOG_SERVER_BACKLOG 4 // max simulataneous connections
+#endif
+
+#ifndef TI_LOG_SERVER_QUEUE_SIZE
+# define TI_LOG_SERVER_QUEUE_SIZE 100
+#endif
+
+/**
+ * All log server state is global mainly because there should only be one log
+ * server.
+ */
+static dispatch_queue_t logDispatchQueue = nil;
+static dispatch_source_t logDispatchSource = nil;
+static int logServerSocket = -1;
+static NSMutableArray* connections = nil;
+static NSMutableArray* logQueue = nil;
+
+/**
+ * Helper function to force logging with NSLog() since we override it. Used only
+ * for reporting log server errors.
+ */
+static void forcedNSLog(NSString* str, ...)
+{
+	va_list args;
+	va_start(args, str);
+	NSString* msg = [[[NSString alloc] initWithFormat:str arguments:args] autorelease];
```

Comment:
I personally don't like to use autorelease in variables that are only used in one method. So I'd rather use
```objc
NSString* msg = [[[NSString alloc] initWithFormat:str arguments:args] retain];
```
and 
```objc
[msg release]
```
after using it.